hnn_core/network_builder.py

```diff
@@ -565,12 +611,27 @@ def get_data_from_neuron(self):
             isoma_py[gid] = {key: rec_i.to_python()
                              for key, rec_i in rec_i.items()}
 
+        lfp_py = list()
+        for elec in self._lfp:
+            lfp_py.append(elec.voltage)
+            elec.reset()
+
+        lfp_arr_py = dict()
+        lfp_times_py = dict()
+        for arr_name, nrn_arr in self._lfp_array.items():
+            voltage_arr = [nrn_arr.voltages.getrow(ii).to_python() for
+                           ii in range(nrn_arr.voltages.nrow())]
+            lfp_arr_py.update({arr_name: voltage_arr})
+            lfp_times_py.update({arr_name: nrn_arr.times})
+            nrn_arr.reset()  # XXX needed?
```

Comment:
This feels like something that should be an `LfpArray` method:

```py
lfp_array.to_python()

data = (..., lfp_array.voltage, lfp_array.times)
```

? As we start exposing more attributes, we should come up with a consistent interface to go from Neuron to Python and to gather from different ranks

Reply:
I've had the same thought as well, but honestly how much is there left recording-wise that we want to expose? Now that we're recording LFP's, it's not a huge leap to consider giving users access to current/voltage from any of segments (considering it has to be recorded anyways).

I suppose one other attribute that comes to mind is synapse specific current flow.